controller/authorize.go

```diff
@@ -67,22 +67,14 @@ func (c *AuthorizeController) Authorize(ctx *app.AuthorizeAuthorizeContext) erro
 		ClientSecret: c.Configuration.GetKeycloakSecret(),
 		Scopes:       scope,
 		Endpoint:     oauth2.Endpoint{AuthURL: authEndpoint, TokenURL: tokenEndpoint},
-		RedirectURL:  rest.AbsoluteURL(ctx.RequestData, "/api/authorize/callback"),
+		RedirectURL:  rest.AbsoluteURL(ctx.RequestData, client.CallbackAuthorizePath()),
 	}
 
 	ctx.ResponseData.Header().Set("Cache-Control", "no-cache")
-	return c.Auth.PerformAuthorize(ctx, oauth, c.Configuration)
+	return c.Auth.AuthCodeURL(ctx, oauth, c.Configuration)
```

Comment:
You still set the location header in the service? As I mentioned a few times it's better if the service (c.Auth.AuthCodeURL(), AuthCodeCallback()) returns URLs only. Then you set the location header in the controller.

Reply:
And same with the Login()(Perform() earlier) method?

Reply:
No. Perform/Login is very old method which handed redirection. Leave it as is.
But new methods like AuthCodeURL() and AuthCodeCallback() should return URLs as we do in https://github.com/fabric8-services/fabric8-auth/blob/master/token/link/link.go#L127 or https://github.com/fabric8-services/fabric8-auth/blob/master/token/link/link.go#L255